File: angular/src/app/model/formateur.ts
import {User} from './user';
import {Role} from './role.enum';
import {Session} from './session';
import {Expertise} from './expertise';

export class Formateur extends User {


  constructor(private _sessions: Session[], private _expertises: Expertise[]) {
    super();
    this.role = Role.FORMATEUR;
  }


  get sessions(): Session[] {
    return this._sessions;
  }

  set sessions(value: Session[]) {
    this._sessions = value;
  }


  get expertises(): Expertise[] {
    return this._expertises;
  }

  set expertises(value: Expertise[]) {
    this._expertises = value;
  }
}

File: java/src/main/java/com/sopra/TPFinal/model/Formateur.java
package com.sopra.TPFinal.model;

import java.util.Set;

import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.OneToMany;

@Entity
@DiscriminatorValue("formateur")
public class Formateur extends User{
	@OneToMany(mappedBy = "key.formateur", fetch = FetchType.LAZY)
	private Set<Expertise> expertises;
	@OneToMany(mappedBy = "key.session", fetch = FetchType.LAZY)
	private Set<Session> sessions;
	
	public Formateur() {
		super();
		super.setRole(Role.FORMATEUR);
	}

	public Set<Expertise> getExpertises() {
		return expertises;
	}

	public void setExpertises(Set<Expertise> expertises) {
		this.expertises = expertises;
	}

	public Set<Session> getSessions() {
		return sessions;
	}

	public void setSessions(Set<Session> sessions) {
		this.sessions = sessions;
	}
	
	
	
}
